internal/core/window.rs

```diff
@@ -678,4 +769,47 @@ pub mod ffi {
         let window = &*(handle as *const WindowRc);
         window.close_popup();
     }
+
+    /// C binding to the set_rendering_notifier() API of Window
+    #[no_mangle]
+    pub unsafe extern "C" fn sixtyfps_windowrc_set_rendering_notifier(
+        handle: *const WindowRcOpaque,
+        callback: extern "C" fn(rendering_state: RenderingState, user_data: *mut c_void),
+        drop_user_data: extern "C" fn(user_data: *mut c_void),
+        user_data: *mut c_void,
+    ) {
+        struct CNotifier {
+            callback: extern "C" fn(rendering_state: RenderingState, user_data: *mut c_void),
+            drop_user_data: extern "C" fn(*mut c_void),
+            user_data: *mut c_void,
+        }
+
+        impl Drop for CNotifier {
+            fn drop(&mut self) {
+                (self.drop_user_data)(self.user_data)
+            }
+        }
+
+        impl RenderingNotifier for CNotifier {
+            fn notify(&mut self, rust_state: super::RenderingState) {
+                let cpp_state = match rust_state {
+                    super::RenderingState::RenderingSetup { .. } => RenderingState::RenderingSetup,
```

Comment:
why is it ok to ignore the contents?

Reply:
Exposing the Rust API to C++ doesn't make sense here. Neither the WebGL bits nor the non-ffi-safe get-proc-address function. The former is never used in C++, the latter isn't typically used: It's common to link against the GL libraries directly, instead of the Rust approach of dynamic linkage for the bindings.

If/when we switch to WGPU, then we could perhaps expose the C API: https://github.com/gfx-rs/wgpu-native, if that's somehow possible.